### DISCUSSION TITLE
Should the semantic conventions module be marked as alpha?

### DISCUSSION THREAD

**Discussion by jkwatson** [open-telemetry/opentelemetry-java#2560] [Q&A]
https://github.com/open-telemetry/opentelemetry-specification/pull/1291 comments that our stability guarantees about telemetry and hence semantic conventions have not yet been determined. Should we be marking the `opentelemetry-semconv` module as `-alpha`? If so, what do we do about the fact that some exporters and pieces of the SDK depend on constants defined in that module?

Opinions wanted!

**Comment by anuraaga** [comment]
Yeah let's mark it alpha - any code we publish that may upgrade and cause a downstream app's build to break needs to be alpha. I presume we will have to deal with removed / renamed keys which would break apps.

**Comment by HaloFour** [comment]
> If so, what do we do about the fact that some exporters and pieces of the SDK depend on constants defined in that module?

Any guidance on what to do in this case?  I have both custom tracing instrumentation as well as a custom exporter that is using some of the constants defined in `SemanticAttributes`, specifically those around exceptions.  I have concerns that anyone that takes a dependency on my modules could observe breaking changes if that semconv module is changed after API/SDK goes 1.0.

**Reply by anuraaga** [reply]
I think realistically you should go ahead and vendor in the constants - in the end they're just Strings and we could say in this case we want to prioritize API compatibility over code reuse. If we came to a point where names changed you could either change or not depending on your own stability guarantees.

This is also something we could do of course but it's currently out of spec for OpenTelemetry for better or worse.

**Reply by jkwatson** [reply]
I also think that the main thing that the exporters depend on is `service.name`, and I would find it incredibly unlikely that will change at this point.

**Reply by HaloFour** [reply]
So the recommendation would be to redefine those constants in my own project rather than taking a dependency on semconv?  At least until that module stabilizes?

**Reply by jkwatson** [reply]
I think it's all up to how much appetite your project has for risk at this point. I personally think many of the constants are very stable at this point, but until the spec has marked things as stable, we don't want to claim stability.

So, if your risk appetite is low, it should be fine to have your own copies of the constants.
